qubes/vm/qubesvm.py

```diff
@@ -1359,6 +1361,8 @@ async def _kill_locked(self):
             raise
 
         # make sure all shutdown tasks are completed
+        self.__waiter = asyncio.get_running_loop().create_future()
+        await self.__waiter
```

Comment:
Please add some timeout. If the event is lost for any reason (including libvirt restart or crash for example), this would wait indefinitely, *while holding the self.startup_lock*.